### PR TITLE
operator install: add must-gather option

### PR DESCRIPTION
**Pull request by chetna14manku**

### CODE REVIEW

File: ocp_utilities/operators.py [hunk 1]
```diff
@@ -391,3 +414,43 @@ def create_catalog_source_from_image(
...
+    """
+    Run must gather command with an option to create target directory.
+
+    Args:
+        must_gather_output_dir (str): Path to base directory where must-gather logs will be stored
+        kubeconfig_path (str): Path to kubeconfig
+        cluster_name (str): Cluster Name
+        operator_name (str): Operator Name
+
+    Returns:
```
Comment (rnetser) [inline ocp_utilities/operators.py:429]: function does not return anything

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by omrirh — APPROVED** [review]

File: ocp_utilities/operators.py [hunk 2]
```diff
@@ -154,72 +158,88 @@ def install_operator(
...
+        )
+    except Exception as ex:
+        LOGGER.error(f"{name} Install Failed. \n{ex}")
+        if must_gather_output_dir:
+            collect_must_gather(
+                must_gather_output_dir=must_gather_output_dir,
+                kubeconfig_path=kubeconfig,
+                cluster_name=cluster_name,
+                product_name=name,
+            )
```
Comment (myakove) [inline ocp_utilities/operators.py:242]: you should raise the Exception in both cases (must_gather_output_dir or not)

File: ocp_utilities/must_gather.py [hunk 3]
```diff
@@ -51,3 +54,41 @@ def run_must_gather(
...
+        must_gather_output_dir (str): Path to base directory where must-gather logs will be stored
+        kubeconfig_path (str): Path to kubeconfig
+        cluster_name (str): Cluster Name for which must-gather will run
+        product_name (str): Product Name installed on given cluster
+    """
+
+    target_dir = os.path.join(must_gather_output_dir, "must-gather", product_name, cluster_name)
+
+    try:
+        if not os.path.exists(kubeconfig_path):
```
Comment (myakove) [inline ocp_utilities/must_gather.py:74]: `run_must_gather` do not force 'kubeconfig', why you force it here?

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_utilities/must_gather.py (hunk 3 above)
Comment (myakove) [reply ocp_utilities/must_gather.py:74]: Not necessarily, if you have export `KUBECONFIG` it will work.

File: ocp_utilities/operators.py (hunk 2 above)
Comment (chetna14manku) [reply ocp_utilities/operators.py:242]: @myakove updated the code

File: ocp_utilities/must_gather.py (hunk 3 above)
Comment (chetna14manku) [reply ocp_utilities/must_gather.py]: Updated the code to use `os.environ.get('KUBECONFIG')` incase `kubeconfig_path` is None

**Review by omrirh — APPROVED** [review]

File: ocp_utilities/must_gather.py [hunk 4]
```diff
@@ -51,3 +54,37 @@ def run_must_gather(
         flag_string = "".join([f" --{flag_name}" for flag_name in flag_names])
         base_command += f" {flag_string}"
     return run_command(command=shlex.split(base_command), check=False)[1]
+
+
+def collect_must_gather(must_gather_output_dir, cluster_name, product_name, kubeconfig_path=None):
+    """
+    Run must-gather command with an option to create target directory using must_gather_output_dir, cluster_name, product_name.
+    (This function can be used for running must-gather incase any product installation fails.)
```
Comment (rnetser) [inline ocp_utilities/must_gather.py:62]: args are described in the `Args` part, no need to repeat here.
please follow doscrting formatting (e.g new line in missing after 1st line and how it should be written)

File: ocp_utilities/must_gather.py [hunk 5]
```diff
@@ -51,3 +54,37 @@ def run_must_gather(
...
+    target_dir = os.path.join(must_gather_output_dir, "must-gather", product_name, cluster_name)
+
+    try:
+        LOGGER.info(f"Prepare must-gather target extracted directory {target_dir}.")
+        Path(target_dir).mkdir(parents=True, exist_ok=True)
+
+        LOGGER.info(f"Collect must-gather for cluster {cluster_name}")
+        run_must_gather(
+            target_base_dir=target_dir,
+            kubeconfig=kubeconfig_path or os.environ.get("KUBECONFIG"),
```
Comment (rnetser) [inline ocp_utilities/must_gather.py:80]: this should not be done here.
`kubeconfig_path` - pass it as is; `run_must_gather` takes care of it.

File: ocp_utilities/operators.py [hunk 6]
```diff
@@ -154,72 +158,89 @@ def install_operator(
         source_image (str, optional): Source image url, If provided install operator from this CatalogSource Image.
         iib_index_image (str, optional): iib index image url, If provided install operator from iib index image.
         brew_token (str, optional): Token to access iib index image registry.
+        must_gather_output_dir (str, optional): Path to base directory where must-gather logs will be stored
+        kubeconfig (str, optional): Path to kubeconfig, required if must_gather_output_dir param provided
```
Comment (rnetser) [inline ocp_utilities/operators.py:162]: not required, see comment above. do not force to pass it.

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**